dynawo/sources/Modeler/Common/DYNModelMulti.cpp

```diff
@@ -1085,6 +1085,9 @@ ModelMulti::updateCalculatedVarForCurves(boost::shared_ptr<curves::CurvesCollect
 }
 
 void ModelMulti::printVariableNames(bool withVariableType) {
+  if (withVariableType) {
```

Comment:
why only with variableType ?

Reply:
I would have thought a method to reset sink of Trace::variables() be called at each call of this function

Reply:
For the first call the log file is totally empty already. So I didn't see why it would be useful to clear the log file.

Reply:
Not an issue to clear something already cleared

Reply:
I would remove the if altogether

Reply:
However we need to have the header only once

Reply:
I would just write the file from scratch each time when calling this method as the variable type is just handled later on but the rest is the same

Reply:
This is what is done, and this is also why we need to rewrite the header

Reply:
I made a mistake, during the first call, the VARIABLES log file contains the Dynawo log header but maybe it's a detail and we can overwrite it as you asked.

Reply:
ok let me try to sum up
- first attempt, withVariableType=false, header already there => we can write directly, or erase everything but rewrite the headers
- second attempt withVariableType=true, file already there => need to erase and rewrite the header.

So either we rewrite evrything each time as suggested or we keep it that way, both works

Reply:
clearVariablesLog();
writeHeader();
... prints
call sequence would do it I would think ?

Reply:
yes

Reply:
So we call 
```
Trace::clearVariablesLogFile();
Trace::printDynawoLogHeader(Trace::variables());
```
 each time we execute ModelMulti::printVariableNames() or not ?

Reply:
I have no strong opinion, let's do it this way

Reply:
Yes it's a log file called only in very specific cases the simpler the better